OpenRA.Game/Renderer.cs

```diff
@@ -267,6 +310,33 @@ public void ReleaseWindowMouseFocus()
 			Window.ReleaseWindowMouseFocus();
 		}
 
+		public void SaveScreenshot(string path)
+		{
+			var src = screenSprite.Sheet.GetData();
+			var srcWidth = screenSprite.Sheet.Size.Width;
+			var destWidth = screenSprite.Bounds.Width;
+			var destHeight = -screenSprite.Bounds.Height;
+			var channelOrder = new[] { 2, 1, 0, 3 };
+
+			ThreadPool.QueueUserWorkItem(_ =>
+			{
+				// Convert BGRA to RGBA
+				var dest = new byte[4 * destWidth * destHeight];
+				for (var y = 0; y < destHeight; y++)
+				{
+					for (var x = 0; x < destWidth; x++)
+					{
+						var destOffset = 4 * (y * destWidth + x);
+						var srcOffset = 4 * (y * srcWidth + x);
+						for (var i = 0; i < 4; i++)
+							dest[destOffset + i] = src[srcOffset + channelOrder[i]];
+					}
+				}
+
+				new Png(dest, destWidth, destHeight).Save(path);
```

Comment:
When I take multiple screenshots - it seems they're all copies of the first screenshot I take?

Reply:
I suspect the first call is buffering the sheet and then returning that buffer each time rather than requesting fresh from the GPU.

Reply:
Fixed. Also fixed transparent pixels when the screenshot extended outside the map bounds.